core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/workspace/runners/BPMBillingProjectDeleteRunnerSpec.scala

```diff
@@ -0,0 +1,464 @@
+package org.broadinstitute.dsde.rawls.monitor.workspace.runners
+
+import bio.terra.workspace.model.{DeleteAzureLandingZoneJobResult, ErrorReport, JobReport}
+import org.broadinstitute.dsde.rawls.TestExecutionContext
+import org.broadinstitute.dsde.rawls.billing.{BillingProjectLifecycle, BillingRepository}
+import org.broadinstitute.dsde.rawls.dataaccess.slick.WorkspaceManagerResourceMonitorRecord
+import org.broadinstitute.dsde.rawls.dataaccess.slick.WorkspaceManagerResourceMonitorRecord.JobType
+import org.broadinstitute.dsde.rawls.dataaccess.{GoogleServicesDAO, SamDAO}
+import org.broadinstitute.dsde.rawls.dataaccess.workspacemanager.WorkspaceManagerDAO
+import org.broadinstitute.dsde.rawls.model.CreationStatuses.DeletionFailed
+import org.broadinstitute.dsde.rawls.model.{RawlsBillingProjectName, RawlsRequestContext, RawlsUserEmail}
+import org.mockito.ArgumentMatchers
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{doReturn, spy, verify, when}
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatestplus.mockito.MockitoSugar
+
+import java.sql.Timestamp
+import java.time.Instant
+import java.util.UUID
+import scala.concurrent.{ExecutionContext, Future}
+
+class BPMBillingProjectDeleteRunnerSpec extends AnyFlatSpec with MockitoSugar with Matchers with ScalaFutures {
+  implicit val executionContext: ExecutionContext = TestExecutionContext.testExecutionContext
+
+  behavior of "initial setup and basic requirements updating billing project status"
+
+  val userEmail = "user@email.com"
+  val billingProjectName: RawlsBillingProjectName = RawlsBillingProjectName("fake_name")
+
+  it should "return a completed status if the billing project name is None" in {
+    val monitorRecord: WorkspaceManagerResourceMonitorRecord =
+      WorkspaceManagerResourceMonitorRecord.forAzureLandingZoneCreate(
+        UUID.randomUUID(),
+        billingProjectName,
+        RawlsUserEmail(userEmail)
+      )
+    val runner = new BPMBillingProjectDeleteRunner(
+      mock[SamDAO],
+      mock[GoogleServicesDAO],
+      mock[WorkspaceManagerDAO],
+      mock[BillingRepository],
+      mock[BillingProjectLifecycle]
+    )
+    whenReady(runner(monitorRecord.copy(billingProjectId = None)))(
+      _ shouldBe WorkspaceManagerResourceMonitorRecord.Complete
+    )
```

Comment:
Verify the args to `updateCreationStatus`?

Reply:
There is no call up `updateCreationStatus` here, because there's no billing project set on the job, and therefore no call to update it.

Reply:
Ah, right!